# AFP data test

In [1]:
import pandas as pd

In [29]:
df = pd.read_csv('data/Missing_Migrants_Global_Figures_allData_1.csv')

In [7]:
df.describe()

Incident year  Number of Dead  Minimum Estimated Number of Missing  \
count   12901.000000    12343.000000                          1296.000000   
mean     2019.025424        2.627562                            18.918210   
std         2.401318        9.765509                            45.960797   
min      2014.000000        0.000000                            -2.000000   
25%      2017.000000        1.000000                             2.000000   
50%      2019.000000        1.000000                             6.000000   
75%      2021.000000        2.000000                            17.000000   
max      2023.000000      750.000000                           750.000000   

       Total Number of Dead and Missing  Number of Survivors  \
count                      12901.000000          2062.000000   
mean                           4.414386            41.098448   
std                           19.971388            99.398680   
min                            1.000000            -2.000000   
25%                            1.000000             4.000000   
50%                            1.000000            13.000000   
75%                            2.000000            38.750000   
max                         1022.000000          1950.000000   

       Number of Females  Number of Males  Number of Children  Source Quality  
count        2327.000000      7954.000000         1568.000000    12900.000000  
mean            2.556081         1.806638            1.970663        3.251395  
std             6.340524         4.719394            6.717124        1.369762  
min             0.000000         0.000000            0.000000        1.000000  
25%             1.000000         1.000000            1.000000        2.000000  
50%             1.000000         1.000000            1.000000        4.000000  
75%             2.000000         1.000000            2.000000        4.000000  
max            94.000000       135.000000          250.000000        5.000000

So we can see in total that the dataset has 12901 entries. 

Let's look at the columns.

In [8]:
# Show all columns
pd.set_option('display.max_columns', None)

In [9]:
df.head(2)

Main ID    Incident ID Incident Type Region of Incident Website Date  \
0  2014.MMP00001  2014.MMP00001      Incident      North America   2014-01-06   
1  2014.MMP00002  2014.MMP00002      Incident      North America   2014-01-12   

   Incident year Reported Month  Number of Dead  \
0           2014        January             1.0   
1           2014        January             1.0   

   Minimum Estimated Number of Missing  Total Number of Dead and Missing  \
0                                  NaN                                 1   
1                                  NaN                                 1   

   Number of Survivors  Number of Females  Number of Males  \
0                  NaN                NaN              1.0   
1                  NaN                NaN              NaN   

   Number of Children               Region of Origin    Cause of Death  \
0                 NaN                Central America  Mixed or unknown   
1                 NaN  Latin America / Caribbean (P)  Mixed or unknown   

  Country of Origin            Migration route  \
0         Guatemala  US-Mexico border crossing   
1           Unknown  US-Mexico border crossing   

                                   Location of death  \
0  Pima Country Office of the Medical Examiner ju...   
1  Pima Country Office of the Medical Examiner ju...   

                                  Information Source             Coordinates  \
0  Pima County Office of the Medical Examiner (PC...  31.650259, -110.366453   
1  Pima County Office of the Medical Examiner (PC...    31.59713, -111.73756   

  UNSD Geographical Grouping  \
0           Northern America   
1           Northern America   

                                       Article title  Source Quality  \
0                                                NaN             5.0   
1  Undocumented Border Crossers 01/01/2014-08/29/...             5.0   

                          URL  
0  http://humaneborders.info/  
1                         NaN

In [10]:
# Let's see what the different migration routes are
df['Migration route'].unique()

array(['US-Mexico border crossing', nan, 'Eastern Mediterranean',
       'English Channel to the UK', 'Caribbean to US',
       'Western Mediterranean', 'Central Mediterranean',
       'Haiti to Dominican Republic', 'Horn of Africa to Yemen crossing',
       'Sahara Desert crossing',
       'Western Africa / Atlantic route to the Canary Islands',
       'Syria to Türkiye', 'Western Balkans', 'Türkiye-Europe land route',
       'Dominican Republic to Puerto Rico', 'Comoros to Mayotte',
       'Venezuela to Caribbean', 'Darien', 'Iran to Türkiye',
       'Italy to France', 'Afghanistan to Iran', 'DRC to Uganda',
       'Central Mediterranean,Sahara Desert crossing',
       'Belarus-EU border', 'Ukraine to Europe',
       'Caribbean to Central America'], dtype=object)

In [30]:
# Save a new dataframe that only represents US-Mexico border
df[df['Migration route']=='US-Mexico border crossing'].to_csv('data/US-Mexico-missing-migrants.csv',index=False)

In [31]:
# Load new data
df = pd.read_csv('data/US-Mexico-missing-migrants.csv')

In [37]:
df.head(2)

Main ID    Incident ID Incident Type Region of Incident Website Date  \
0  2014.MMP00001  2014.MMP00001      Incident      North America   2014-01-06   
1  2014.MMP00002  2014.MMP00002      Incident      North America   2014-01-12   

   Incident year Reported Month  Number of Dead  \
0           2014        January             1.0   
1           2014        January             1.0   

   Minimum Estimated Number of Missing  Total Number of Dead and Missing  \
0                                  NaN                                 1   
1                                  NaN                                 1   

   Number of Survivors  Number of Females  Number of Males  \
0                  NaN                NaN              1.0   
1                  NaN                NaN              NaN   

   Number of Children               Region of Origin    Cause of Death  \
0                 NaN                Central America  Mixed or unknown   
1                 NaN  Latin America / Caribbean (P)  Mixed or unknown   

  Country of Origin            Migration route  \
0         Guatemala  US-Mexico border crossing   
1           Unknown  US-Mexico border crossing   

                                   Location of death  \
0  Pima Country Office of the Medical Examiner ju...   
1  Pima Country Office of the Medical Examiner ju...   

                                  Information Source             Coordinates  \
0  Pima County Office of the Medical Examiner (PC...  31.650259, -110.366453   
1  Pima County Office of the Medical Examiner (PC...    31.59713, -111.73756   

  UNSD Geographical Grouping  \
0           Northern America   
1           Northern America   

                                       Article title  Source Quality  \
0                                                NaN             5.0   
1  Undocumented Border Crossers 01/01/2014-08/29/...             5.0   

                          URL  
0  http://humaneborders.info/  
1                         NaN

In [38]:
# Check that the number of dead matches what we're expecting
df['Number of Dead'].sum()

4467.0